Pointers/Counter.cpp

```diff
@@ -0,0 +1,34 @@
+#include "Counter.h"
+#include <iostream>
+
+using std::cout;
+using std::endl;
+
+Counter::Counter(uint32_t startValue) : m_counter(startValue) {
+	cout << "Initialize refcount" << endl;
+	printState(); 
+}
+
+Counter::~Counter() {
```

Comment:
Why do you need to implement a destructor here? Once you destroy the object no one can access m_counter